From d0c86388822fd40d804c8ed787e2ad600176be13 Mon Sep 17 00:00:00 2001
From: Philippe Normand <philn@igalia.com>
Date: Sat, 20 Aug 2022 16:57:27 +0100
Subject: [PATCH] openh264enc: Fix constrained-high encoding

constrained-high is high without B-frames, there is no EProfileIdc for this, so
assume high instead of hitting an assert down the line.
---
 .../gst-plugins-bad/ext/openh264/gstopenh264enc.cpp        | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/ext/openh264/gstopenh264enc.cpp b/ext/openh264/gstopenh264enc.cpp
index 30af8e2677..3328d01d99 100644
--- a/ext/openh264/gstopenh264enc.cpp
+++ b/ext/openh264/gstopenh264enc.cpp
@@ -711,11 +711,12 @@ gst_openh264enc_get_profile_from_caps (GstCaps *outcaps, GstCaps *allowed_caps)
 
   gst_structure_set (s, "profile", G_TYPE_STRING, profile, NULL);
   if (!g_strcmp0 (profile, "constrained-baseline") ||
-       !g_strcmp0 (profile, "baseline"))
+      !g_strcmp0 (profile, "baseline"))
     return PRO_BASELINE;
-   else if (!g_strcmp0 (profile, "main"))
+  else if (!g_strcmp0 (profile, "main"))
     return PRO_MAIN;
-   else if (!g_strcmp0 (profile, "high"))
+  else if (!g_strcmp0 (profile, "high") ||
+      !g_strcmp0 (profile, "constrained-high"))
     return PRO_HIGH;
 
   g_assert_not_reached ();
-- 
2.37.1

